samples/cellular/nrf_cloud_multi_service/src/fota_support.c

```diff
@@ -7,18 +7,24 @@
 #include <zephyr/logging/log.h>
 #include <zephyr/sys/reboot.h>
 #include <zephyr/logging/log_ctrl.h>
+#include <modem/lte_lc.h>
 #include <net/nrf_cloud.h>
 
 #include "fota_support.h"
 
-LOG_MODULE_REGISTER(fota_support, CONFIG_MQTT_MULTI_SERVICE_LOG_LEVEL);
+LOG_MODULE_REGISTER(fota_support, CONFIG_MULTI_SERVICE_LOG_LEVEL);
 
 /* Called from nRF Cloud event handler in connection.c */
-void on_fota_downloaded(void)
+void fota_reboot(const unsigned int delay_s, const bool error)
```

Comment:
Can you please create a new `on_fota_downloaded` function that takes no arguments, and then calls `fota_reboot(FOTA_REBOOT_DELAY_S, false);`?

Then make `FOTA_REBOOT_DELAY` a KConfig for the sample, or do something else so that it is defined here (speaking of which, it looks like you've defined it both in `cloud_connection.c` and in `fota_support_coap.c`)

Then the call to `fota_reboot` in `cloud_connection.c` can go back to being `on_fota_downloaded`, which IMO enforces better separation of concerns without actually changing your work too much